packages/compass-schema/src/stores/schema-export-reducer.ts

```diff
@@ -150,54 +155,118 @@ async function getSchemaByFormat({
   return JSON.stringify(schema, null, 2);
 }
 
+export const downloadSchema = (): SchemaThunkAction<void> => {
+  return (dispatch, getState, { track, connectionInfoRef }) => {
+    const {
+      schemaExport: { exportedSchema, exportFormat, filename },
+    } = getState();
+    if (!exportedSchema) return;
+    try {
+      const blob = new Blob([exportedSchema], {
+        type: 'application/json',
+      });
+      const url = window.URL.createObjectURL(blob);
+
+      const link = document.createElement('a');
+      link.href = url;
+      link.download = filename || 'export.json';
+      link.click();
+      setTimeout(() => {
+        window.URL.revokeObjectURL(url);
+        link.remove();
+      }, 0);
+      dispatch(trackSchemaExported);
```

Comment:
Still not doing anything because you are passing the thunk action creator to dispatch, not thunk action itself

```suggestion
      dispatch(trackSchemaExported());
```

Reply:
oh my! thank you for being so thorough with the reviews, I really need to slow down and pay more attention